src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -1053,19 +1090,24 @@ const put = async (frameworkName, config, rawConfig) => {
     config,
     rawConfig,
   );
-  // generate image pull secret
+  // generate the image pull secret definition
   const auths = Object.values(config.prerequisites.dockerimage)
     .filter((dockerimage) => dockerimage.auth != null)
     .map((dockerimage) => dockerimage.auth);
   const dockerSecretDef = auths.length
     ? getDockerSecretDef(frameworkName, auths)
     : null;
 
-  // generate job config secret
+  // generate the job config secret definition
   const configSecretDef = config.secrets
     ? getConfigSecretDef(frameworkName, config.secrets)
     : null;
 
+  // create an application token
+  const token = await tokenModel.create(userName, true);
```

Comment:
What if:

```
1. the write merger is down 
2. the token is created here
3. the job cannot be submitted to write merger
```

This token will never be deleted or revoked.

Reply:
Better to give each token a name or a label related to the job thus we can recycle them in the watchdog

Reply:
> Better to give each token a name or a label related to the job thus we can recycle them in the watchdog (maybe implemented in the future)

Added a TODO item here, will resolve this later